pkg/grpcx/credentials.go

```diff
@@ -0,0 +1,49 @@
+package grpcx
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/kubeshop/botkube/pkg/config"
+)
+
+// ClientTransportCredentials returns gRPC client transport credentials based on the provided configuration.
+func ClientTransportCredentials(log logrus.FieldLogger, cfg config.GRPCServer) (credentials.TransportCredentials, error) {
+	if cfg.DisableSecurity {
+		log.Warn("Server CA certificate is not provided. Using insecure gRPC connection...")
```

Comment:
It could be misleading; the cert can be provided, but simply flag can disable it. Please update the msg 👍 Also, 'insecure' may be understood as not only encryption is disabled but also auth, so I would update that as well, thanks!